scm-core/src/main/java/sonia/scm/repository/api/PushCommandBuilder.java

```diff
@@ -38,44 +36,41 @@
 import java.io.IOException;
 import java.net.URL;
 
-//~--- JDK imports ------------------------------------------------------------
-
 /**
  * The push command push changes to a other repository.
  *
  * @author Sebastian Sdorra
  * @since 1.31
  */
-public final class PushCommandBuilder
-{
+public final class PushCommandBuilder {
 
-  /**
-   * the logger for PushCommandBuilder
-   */
-  private static final Logger logger =
-    LoggerFactory.getLogger(PushCommandBuilder.class);
+  private static final Logger logger = LoggerFactory.getLogger(PushCommandBuilder.class);
 
-  //~--- constructors ---------------------------------------------------------
+  private final PushCommand command;
+  private final PushCommandRequest request = new PushCommandRequest();
+
+  PushCommandBuilder(PushCommand command) {
+    this.command = command;
+  }
 
   /**
-   * Constructs a new PushCommandBuilder.
+   * Set basic authentication for request
    *
-   * @param command implementation of the {@link PushCommand}
+   * @param username username
+   * @return this builder instance.
+   * @since 2.22.0
    */
-  PushCommandBuilder(PushCommand command)
-  {
-    this.command = command;
+  public PushCommandBuilder withBasicAuth(String username, String password) {
```

Comment:
Why basic auth? Is the combination only usable for http basic authentication? Why not `withUsernameAndPassword` or `withCredentials`?